src/Nethermind/Nethermind.Merge.Plugin/Synchronization/MergeBlockDownloader.cs

```diff
@@ -225,16 +227,19 @@ bool HasMoreToSync(out BlockHeader[]? headers, out int headersToRequest)
                     {
                         // covering edge case during fastSyncTransition when we're trying to SuggestBlock without the state
                         bool headIsGenesis = _blockTree.Head?.IsGenesis ?? false;
-                        bool toBeProcessedIsNotBlockOne = currentBlock.Number > 1;
+                        bool toBeProcessedIsNotBlockOne = currentBlock.Number > (bestSuggestedBlock + 1);
```

Comment:
@LukaszRozmej 
1. `bestSuggestedBlock` is added because we trigger `isFastSyncTransition` branch even if we have state for processing during sequential block downloading. 
2. The `isFastSyncTransition` branch was changing `downloadReceipts` flag without doing actual late receipts downloading, which always raised exception at [288](https://github.com/NethermindEth/nethermind/pull/6413/files#diff-bc1a38c2ac3bf274bd3b4759d4f132c287ec6a4db33035d7f74ea7c4a28d68b2R288), so I changed it to be sure.

Reply:
But we shouldn't do fast sync if we already have state (on post merge chains), so I don't get it.

Reply:
This code path is executed when we need to download blocks before the block with state, but because the state is downloaded, the code will command to download AND process. But it's not possible for pre-state blocks, so we switch back to not-processing path (which means downloading receipts for them)
But full sync triggers this code anyway, and I made it not to switch to receipts downloading. And also switching to receipts downloading actually asks for receipts that were not loaded.
I've added a comment also

Reply:
@LukaszRozmej ^